### PR TITLE
Add missing dependencies

### LINKED ISSUE #2565: Bug: Some modules are not resolved in Yarn PnP mode.
Lexical version: 0.3.5

## Steps To Reproduce

1. Clone repository
```
git clone https://github.com/flex-kwoncheol/lexical-pnp.git
```

2. Run Next.js app
```
yarn install
yarn dev
```

3. Visit `http://localhost:3000`

You will encounter the following error:
```
error - .yarn/__virtual__/@lexical-clipboard-virtual-1e81a771a8/0/cache/@lexical-clipboard-npm-0.3.6-3dd2954f89-c4c5e4699b.zip/node_modules/@lexical/clipboard/LexicalClipboard.dev.js:10:0
Module not found: Can't resolve '@lexical/list'

Import trace for requested module:
./.yarn/__virtual__/@lexical-clipboard-virtual-1e81a771a8/0/cache/@lexical-clipboard-npm-0.3.6-3dd2954f89-c4c5e4699b.zip/node_modules/@lexical/clipboard/LexicalClipboard.js
./.yarn/__virtual__/@lexical-plain-text-virtual-502f472ec8/0/cache/@lexical-plain-text-npm-0.3.6-28a6eff6a5-6bde962fc9.zip/node_modules/@lexical/plain-text/LexicalPlainText.dev.js
./.yarn/__virtual__/@lexical-plain-text-virtual-502f472ec8/0/cache/@lexical-plain-text-npm-0.3.6-28a6eff6a5-6bde962fc9.zip/node_modules/@lexical/plain-text/LexicalPlainText.js
./.yarn/__virtual__/@lexical-react-virtual-3119d8ff1f/0/cache/@lexical-react-npm-0.3.6-bde13fffbb-9eb931833a.zip/node_modules/@lexical/react/LexicalPlainTextPlugin.dev.js
./.yarn/__virtual__/@lexical-react-virtual-3119d8ff1f/0/cache/@lexical-react-npm-0.3.6-bde13fffbb-9eb931833a.zip/node_modules/@lexical/react/LexicalPlainTextPlugin.js
./pages/index.tsx

https://nextjs.org/docs/messages/module-not-found
wait  - compiling /_error (client and server)...
error - .yarn/__virtual__/@lexical-clipboard-virtual-1e81a771a8/0/cache/@lexical-clipboard-npm-0.3.6-3dd2954f89-c4c5e4699b.zip/node_modules/@lexical/clipboard/LexicalClipboard.dev.js:10:0
Module not found: Can't resolve '@lexical/list'

Import trace for requested module:
./.yarn/__virtual__/@lexical-clipboard-virtual-1e81a771a8/0/cache/@lexical-clipboard-npm-0.3.6-3dd2954f89-c4c5e4699b.zip/node_modules/@lexical/clipboard/LexicalClipboard.js
./.yarn/__virtual__/@lexical-plain-text-virtual-502f472ec8/0/cache/@lexical-plain-text-npm-0.3.6-28a6eff6a5-6bde962fc9.zip/node_modules/@lexical/plain-text/LexicalPlainText.dev.js
./.yarn/__virtual__/@lexical-plain-text-virtual-502f472ec8/0/cache/@lexical-plain-text-npm-0.3.6-28a6eff6a5-6bde962fc9.zip/node_modules/@lexical/plain-text/LexicalPlainText.js
./.yarn/__virtual__/@lexical-react-virtual-3119d8ff1f/0/cache/@lexical-react-npm-0.3.6-bde13fffbb-9eb931833a.zip/node_modules/@lexical/react/LexicalPlainTextPlugin.dev.js
./.yarn/__virtual__/@lexical-react-virtual-3119d8ff1f/0/cache/@lexical-react-npm-0.3.6-bde13fffbb-9eb931833a.zip/node_modules/@lexical/react/LexicalPlainTextPlugin.js
./pages/index.tsx

https://nextjs.org/docs/messages/module-not-found
error - Error: @lexical/html tried to access @lexical/selection, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: @lexical/selection
Required by: @lexical/html@virtual:1e81a771a8734ba5750c7404f82baa6fa9911708d10cb918456a7b273cabbcc1025dcc46ccbc52f9f8d2753736a14486eeb281186aaf80a2b7d8e3e69e441a52#npm:0.3.6 (via /Users/kwoncheolshin/flex/lexical-pnp/.yarn/__virtual__/@lexical-html-virtual-d5c2940638/0/cache/@lexical-html-npm-0.3.6-019026dd9d-bf0f455a20.zip/node_modules/@lexical/html/)
```

Link to code example: https://github.com/flex-kwoncheol/lexical-pnp

## The current behavior

Some modules are not resolved in Yarn PnP mode.

## The expected behavior

It also works well in yarn pnp mode.

### PR DESCRIPTION
**Pull request by thegreatercurve**
Closes #2565

### CODE REVIEW

File: packages/lexical-clipboard/package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   },
   "dependencies": {
     "@lexical/utils": "0.3.6",
+    "@lexical/list": "0.3.6",
```
Comment (acywatson) [inline packages/lexical-clipboard/package.json:19]: I know why we have to do this right now, but this is dependency doesn't really make sense. We need to move that special case list logic to the List package somehow.

I'm ok to ship this now to fix the issues, but can we make a follow up issue?

File: packages/lexical-html/package.json [hunk 2]
```diff
@@ -17,5 +17,8 @@
     "type": "git",
     "url": "https://github.com/facebook/lexical",
     "directory": "packages/lexical-html"
+  },
+  "devDependencies": {
+    "@lexical/selection": "^0.3.6"
```
Comment (acywatson) [inline packages/lexical-html/package.json:22]: Why is this a devDependency?

Comment (thegreatercurve) [reply packages/lexical-html/package.json:22]: Changed

File: packages/lexical-clipboard/package.json (hunk 1 above)
Comment (thegreatercurve) [reply packages/lexical-clipboard/package.json:19]: Added an issue here #2590

**Pull request merged**